src/contexts/PreferencesContext.tsx

```diff
@@ -147,15 +147,29 @@ export const PreferencesProvider = ({
     preferenceState: T[],
     setPreferenceState: React.Dispatch<React.SetStateAction<T[]>>,
     preferenceKey: string,
-    existingItemIndex: number,
-    newItem: T
+    existingItemIndex: number | number[],
+    newItem: T | T[]
   ) {
     let newFavorites = [...preferenceState];
-    if (existingItemIndex >= 0) {
-      newFavorites.splice(existingItemIndex, 1);
-    } else {
-      newFavorites = [...newFavorites, newItem];
+    if (Array.isArray(existingItemIndex) && Array.isArray(newItem)) {
+      existingItemIndex.forEach((itemIndex, index) => {
+        if (itemIndex >= 0) {
+          newFavorites.splice(itemIndex, 1);
+        } else {
+          newFavorites = [...newFavorites, newItem[index]];
+        }
+      });
+    } else if (
+      typeof existingItemIndex === 'number' &&
+      !Array.isArray(newItem)
```

Comment:
This if/else is dangerous because it does nothing in several cases. I think it would be better to have each if condition check a single condition (e.g. `if (Array.isArray(existingItemIndex)`) and then have nested ifs for multiple conditions. That would allow us to handle every else case and throw errors for invalid cases. 

But before making any additional changes, let's discuss how these preferences are being handled at a conceptual level, and if the scalar/array duality is really necessary. It would be ideal to delegate this kind of complexity to the backend if possible.